### PR TITLE
Updates modeling docs for VLANs.

### LINKED ISSUE #5608: Docs Update: VLAN Modeling with M2M Addition
### Proposed Changes

With Nautobot 2.2 introducing M2M of VLAN to Locations, it is necessary to recommend a suggested usage of the environments and what to use.

### Justification

Networking is already a snowflake enough. We don't necessarily need to have snowflakes of how to model things.

### PR DESCRIPTION
**Pull request by jvanderaa**
# Closes #5608 
# What's Changed

Updated the docs to include how to model VLANs with the recent addition of M2M for VLANs to Locations in Nautobot 2.2.
# Screenshots

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [x] Attached Screenshots, Payload Example
- [x] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [x] Example App Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/docs/user-guide/core-data-model/ipam/vlan.md [hunk 1]
```diff
@@ -18,3 +18,17 @@ As with prefixes, each VLAN may also be assigned a functional role. Prefixes and
...
+
+## Modeling VLANs
+
+With the update to Nautobot 2.2 that introduced the ability to have a single VLAN be associated to multiple locations, there needs to be some _recommended_ methods to model VLANs with the new functionality.
+
+You should be looking to model VLANs in the same way that the layer 2 functionality is set up.
+
+### Multiple Locations Using Same VLAN ID
+
+In many organizations the use of cookiecutter like VLAN architecture is leveraged. In that a location may have VLAN30 be for the same purpose (ie Wireless, User, Voice) at each location. In this scenario, you should have a VLAN30 for each location. Do **not** create a single VLAN and then associate it with multiple locations.
```
Comment (gsnider2195) [inline nautobot/docs/user-guide/core-data-model/ipam/vlan.md:30]: Can we provide some justification for this recommendation?

File: nautobot/docs/user-guide/core-data-model/ipam/vlan.md [hunk 2]
```diff
@@ -18,3 +18,17 @@ As with prefixes, each VLAN may also be assigned a functional role. Prefixes and
...
+
+### Multiple Locations, Stretched Layer 2
+
+Where you would want to take into account of the Many to Many capability where you can have a single VLAN that is associated with multiple locations is where the layer 2 environment is in fact stretched and devices within the VLAN are able to communicate via layer 2 mechanisms. Such designs include a campus network where you do have multiple locations but a shared VLAN across all of them for say wireless networks. Or in a data center fabric where the layer 2 is in fact stretched across the sites.
```
Comment (gsnider2195) [inline nautobot/docs/user-guide/core-data-model/ipam/vlan.md:34]: Perhaps?

```suggestion
You would want to have a single VLAN that is associated with multiple Locations when the layer 2 environment is in fact stretched and devices within the VLAN are able to communicate via layer 2 mechanisms. Such designs include a campus network where you do have multiple Locations but a shared VLAN across all of them, for example wireless networks. Or in a data center fabric where the layer 2 is in fact stretched across the sites.
```

File: nautobot/docs/user-guide/core-data-model/ipam/vlan.md (hunk 1 above)
Comment (gsnider2195) [reply nautobot/docs/user-guide/core-data-model/ipam/vlan.md:30]: Maybe something like this?

```suggestion
In many organizations the use of cookiecutter like VLAN architecture is leveraged. In that a location may have VLAN30 be for the same purpose (ie Wireless, User, Voice) at each location. In this scenario, you should model a distinct VLAN30 in Nautobot for each Location. Creating a single VLAN and then associating it with multiple Locations would not accurately reflect the state of the layer 2 network.
```

File: changes/5608.documentation [hunk 3]
```diff
@@ -0,0 +1 @@
+Updated VLAN documentation for proper modeling of VLAN M2M.
```
Comment (glennmatthews) [inline changes/5608.documentation:1]: Perhaps?

```suggestion
Updated VLAN documentation with a recommendation for modeling of VLANs with respect to Locations.
```

**Review by gsnider2195 — APPROVED** [review]

**Pull request merged**